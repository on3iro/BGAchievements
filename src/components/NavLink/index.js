/**
  * Renders a single NavLink item containing an li and a preformatted react-router Link
  *
  * @namespace Navigation.NavLink
  * @memberOf Navigation
  */

import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { NavLink } from 'react-router-dom'

import { surpressNoUsedVarsLinterError } from 'shared/helperFunctions'

export const Li = styled.li`
  display: inline-block;
  height: 100%;

  text-align: center;

  &:hover a {
    color: ${props => props.theme.main};
  }

  @media (max-width: 900px) {
    display: block;
  }
`

export const RestyledLink = styled(NavLink)`
  display: inline-block;
  height: 100%;
  padding: 5px 25px;

  color: ${props => props.theme.noneWhite};
  font-size: 1.4rem;
  font-family: ${props => props.theme.primaryFont};
  text-decoration: none;

  outline: 0;

  &.active {
    color: ${props => props.theme.main};
  }
`

const StyledNavLink = props => {
  // pull props off to avoid react-unknown-prop error
  const { icon, iconHover, secondary, ...rest } = props
  surpressNoUsedVarsLinterError({ icon, iconHover, secondary })

  return (
    <Li>
      <RestyledLink
        {...rest}
      >
        {props.children}
      </RestyledLink>
    </Li>
  )
}

StyledNavLink.propTypes = {
  to: PropTypes.string.isRequired,
  children: PropTypes.any.isRequired,
  icon: PropTypes.string,
  iconHover: PropTypes.string,
  secondary: PropTypes.bool
}

export default StyledNavLink
