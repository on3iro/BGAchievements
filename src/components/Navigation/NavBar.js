/**
  * Renders a container NavBar for the Navigation
  *
  * @namespace Navigation.NavBar
  * @memberOf Navigation
  */

import styled from 'styled-components'

export default styled.ul`
  padding: 0;
  margin: 0;
  height: 100%;

  list-style-type: none;
`
