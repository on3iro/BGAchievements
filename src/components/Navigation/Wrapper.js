import styled from 'styled-components'

export default styled.nav`
  margin: 12px 50px 15px 0;
  height: 100%;

  text-align: right;
`
