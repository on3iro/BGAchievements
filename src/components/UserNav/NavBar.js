/**
  * Renders a container NavBar for the Navigation
  *
  * @namespace Navigation.NavBar
  * @memberOf Navigation
  */

import styled from 'styled-components'

export default styled.ul`
  margin-top: 13px;
  margin-right: 80px;
  padding: 0;
  height: 100%;

  text-align: center;
  list-style-type: none;

  li {
    &:last-of-type {
      margin-right: 0;
    }
  }
`
