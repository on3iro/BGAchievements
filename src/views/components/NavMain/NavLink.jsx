// @flow

import React from 'react'
import styled from 'styled-components'
import { NavLink } from 'react-router-dom'

const LinkWithActiveClass = props => (
  <NavLink {...props} activeClassName='selected' />
)

const StyledNavLink = styled(LinkWithActiveClass)`
  display: block;
  padding: 10px;

  color: ${props => props.theme.white};

  cursor: pointer;
  text-decoration: none;

  &.selected {
    background: ${props => props.theme.bgPrimary};
    color: ${props => props.theme.primary};

    &:visited {
      color: ${props => props.theme.primary};
    }
  }

  &:visited {
    color: inherit;
  }
`

export {
  StyledNavLink as default
}
