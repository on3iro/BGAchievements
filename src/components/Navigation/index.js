/**
  * Renders a Navigation with react-router Links
  *
  * @namespace Navigation
  */

import React from 'react'
import PropTypes from 'prop-types'

import Wrapper from './Wrapper'
import NavBar from './NavBar'
import NavLink from 'components/NavLink'

const Navigation = props => {
  return (
    <Wrapper>
      <NavBar>
        {
          props.isLoggedIn &&
            <NavLink to='/home'>
              Home
            </NavLink>
        }
      </NavBar>
    </Wrapper>
  )
}

Navigation.propTypes = {
  isLoggedIn: PropTypes.bool
}

export default Navigation
