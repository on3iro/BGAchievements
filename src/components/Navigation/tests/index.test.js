import React from 'react'
import { shallow } from 'enzyme'

import Navigation from '../index'
import NavBar from '../NavBar'
import NavLink from 'components/NavLink'

describe('<Navigation />', () => {
  it('Should render NavBar', () => {
    const wrapper = shallow(<Navigation />)
    expect(wrapper.find(NavBar).length).toMatchSnapshot()
  })

  it('Should render NavLinks', () => {
    const wrapper = shallow(<Navigation />)
    expect(wrapper.find(NavLink).length).toMatchSnapshot()
  })
})
