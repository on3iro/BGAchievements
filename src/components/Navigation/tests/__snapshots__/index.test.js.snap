exports[`<Navigation /> Should render NavBar 1`] = `1`

exports[`<Navigation /> Should render NavLinks 1`] = `2`
